### ISSUE TITLE
Tracking issue: Adding directives to WooCommerce blocks

### ISSUE THREAD

**Issue by DAreRodz** [woothemes/woocommerce#42486]
This aims to be the tracking issue to follow the progress of including the [Interactivity API](https://github.com/WordPress/block-hydration-experiments/) in some WooCommerce blocks, especially the Product block. As WooCommerce blocks could potentially benefit from this new API, we want to understand what is needed in order to implement some of the blocks using it.

Note that the mentioned API will ideally be included in Gutenberg and WordPress core at some point. However, to start experimenting as soon as possible, we are moving the needed scripts into the repo so we can start testing the API now.

## Experiments
-  https://github.com/woocommerce/woocommerce-blocks/pull/9926
- [**Stackblitz - Interactivity API demo: filters with multiple product queries**](https://stackblitz.com/edit/demo-filters-multiple-queries)

## Done
- https://github.com/woocommerce/woocommerce-blocks/pull/8220
- https://github.com/woocommerce/woocommerce-blocks/pull/8316
- https://github.com/woocommerce/woocommerce-blocks/pull/8447
- https://github.com/woocommerce/woocommerce-blocks/pull/9924
- https://github.com/woocommerce/woocommerce-blocks/pull/10338
- https://github.com/woocommerce/woocommerce-blocks/pull/10200
- Migrate Add to Cart button block to use the Interactivity API. https://github.com/woocommerce/woocommerce-blocks/pull/10006
- https://github.com/woocommerce/woocommerce-blocks/issues/11065

## In progress

- [x] https://github.com/woocommerce/woocommerce/issues/42255

## Next

- [ ] Provide APIs to handle different loading states and error handling.
- [ ] Define and add the opt-in mechanism for client-side navigation.
- [ ] Polish and finish the needed directives.
- [ ] Add test coverage.
- [ ] Review the bundling process and ensure we are shipping the Interactivity API correctly.

**Comment by DAreRodz** [comment]
> ~I asked @SantosGuillamot to update https://github.com/woocommerce/woocommerce-blocks/pull/8026 title and description. It should be ready to review now. 😊~
> 
> - ~https://github.com/woocommerce/woocommerce-blocks/pull/8026~

EDIT: the PR was moved to https://github.com/woocommerce/woocommerce-blocks/pull/8220. This is now the one to be reviewed.

- https://github.com/woocommerce/woocommerce-blocks/pull/8220

**Comment by DAreRodz** [comment]
I've made some progress with the experiment and recorded a video showing how the simple price filter block behaves now. You can see it in the PR description.

- https://github.com/woocommerce/woocommerce-blocks/pull/8221

**Comment by luisherranz** [comment]
Let's try using `data-woo-` to see how it feels like.

Ref: https://github.com/WordPress/block-hydration-experiments/issues/132

**Comment by luisherranz** [comment]
Let's try using `wooStore` instead of `woox` as well.

Ref: https://github.com/WordPress/block-hydration-experiments/issues/136

**Comment by DAreRodz** [comment]
PR https://github.com/woocommerce/woocommerce-blocks/pull/8220 is now merged. 🚀 

- https://github.com/woocommerce/woocommerce-blocks/pull/8220

**Comment by DAreRodz** [comment]
I've added a link to an experiment that shows how to use the Interactivity API for filters with multiple product query blocks.

Remember that it's still WIP 🚧, as it depends on https://github.com/WordPress/block-hydration-experiments/pull/139 to be finished and merged.

I'll work on documenting everything as well once the code is ready.

**Comment by DAreRodz** [comment]
I've started working on replacing all the `wp-` prefixes with `woo` ones. It's still a draft because it depends on https://github.com/WordPress/block-hydration-experiments/pull/142 to be approved and merged.

- https://github.com/woocommerce/woocommerce-blocks/pull/8316

Also, I've updated both demos to use these changes, so now directives start with `data-woo` instead of `wp`.

**Comment by DAreRodz** [comment]
BTW, I forgot to mention that the [Stackblitz - Interactivity API demo: filters with multiple product queries](https://stackblitz.com/edit/demo-filters-multiple-queries) demo is working now.

I still have to document it, but you can take a look at the code already if you want. 😄

**Comment by DAreRodz** [comment]
Another update:

- The [Stackblitz - Interactivity API demo: filters with multiple product queries](https://stackblitz.com/edit/demo-filters-multiple-queries) demo has been updated to remove some preact components I was using to simulate SSR and use plain HTML instead. I hope that makes the code clearer.
- I've started working on store serialization and hydration in the main repo (https://github.com/WordPress/block-hydration-experiments/pull/147). Once merged, I'll port those changes to this repo.

**Comment by DAreRodz** [comment]
Quick update:

The changes for the directives prefix have been recently merged:
- https://github.com/woocommerce/woocommerce-blocks/pull/8316

The PR that adds store serialization and hydration (among other things) is ready to review:
- https://github.com/woocommerce/woocommerce-blocks/pull/8447

**Comment by DAreRodz** [comment]
Forgot to mention that I also updated woocommerce/woocommerce-blocks#8221 (the Simple Price Filter demo) with the latest changes in woocommerce/woocommerce-blocks#8447. 🙂

**Comment by DAreRodz** [comment]
Sorry for the long silence here. 😅 

A while ago, I started working on woocommerce/woocommerce-blocks#8524 to make the Filter blocks start using the Interactivity API, which will close issue woocommerce/woocommerce-blocks#8456.

- https://github.com/woocommerce/woocommerce-blocks/pull/8524

It got stale because it depended on https://github.com/WordPress/block-interactivity-experiments/pull/163, which is already implemented and merged. I'll move those changes to woocommerce/woocommerce-blocks#8524 and continue working on the remaining tasks there. 🙂

**Comment by DAreRodz** [comment]
I've finished updating the Interactivity API code inside WooCommerce blocks. This is the PR for these changes, ready for review.

- https://github.com/woocommerce/woocommerce-blocks/pull/9924

I also created a new PR for the Simple Price Filter block, showcasing how to use the updated Interactivity API in blocks. The old one is closed.

- https://github.com/woocommerce/woocommerce-blocks/pull/9926

**Comment by DAreRodz** [comment]
I've opened a PR implementing server-side-rendered updates for interactive regions during client-side navigation.

- https://github.com/woocommerce/woocommerce-blocks/pull/10200

I also updated the Tracking Issue description to reflect the current status.

**Comment by DAreRodz** [comment]
The Woo team has started working on the Filter blocks and created their own Epic:

- https://github.com/woocommerce/woocommerce/issues/42255

I decided to close the related issue and PR we created a while ago, as the work there was unfinished.

- woocommerce/woocommerce-blocks#8456
- https://github.com/woocommerce/woocommerce-blocks/pull/8524

**Comment by DAreRodz** [comment]
Heads up, we have started working on the new `store()` API, as proposed in https://github.com/WordPress/gutenberg/discussions/53586. The work is being tracked here:

- https://github.com/woocommerce/woocommerce-blocks/issues/11065

**Comment by DAreRodz** [comment]
I forgot to announce that the new Interactivity `store()` API was finally implemented, and interactive blocks were migrated to the new API.

- https://github.com/woocommerce/woocommerce-blocks/issues/11065

Now, the Frontend Filters improvements can benefit from these changes. 😊

**Comment by luisherranz** [comment]
Let's close this old tracking issue. The latest work regarding the Interactivity API is here:

- https://github.com/woocommerce/woocommerce/issues/54259
